internal/task/task/bs/format_stop.go

```diff
@@ -0,0 +1,218 @@
+	return func(ctx *context.Context) error {
+		if len(*containerId) < 0 {
+			return task.ERR_SKIP_TASK
+		}
+		return nil
+	}
+}
+
+// expression 生成要删除或者添加的fstab的sed表达式
+func (s *deleteFSTab) expression(express2del, express2add *string) step.LambdaType {
+	return func(ctx *context.Context) error {
+		*express2del = fmt.Sprintf("/UUID=%s/d", *s.oldUUID)
+		*express2add = fmt.Sprintf("$aUUID=%s  %s  ext4  rw,errors=remount-ro  0  0  # %s",
+			s.uuid, s.mountPoint, WARNING_EDIT)
+		return nil
+	}
+}
+
+// execute 执行删除fstab记录
+func (s *deleteFSTab) execute(ctx *context.Context) error {
+	var express2del, express2add string
+	curveadm := s.curveadm
+	now := time.Now().Format("2006-01-02")
+	steps := []task.Step{}
+
+	var success bool
+	var copyOut string
+	steps = append(steps, &step.CopyFile{ // backup fstab
+		Source:      os.GetFSTabPath(),
+		Dest:        fmt.Sprintf("%s-%s.backup", os.GetFSTabPath(), now),
+		NoClobber:   true,
+		Out:         &copyOut,
+		ExecOptions: curveadm.ExecOptions(),
+	})
+	steps = append(steps, &step.ListBlockDevice{ // uuid for device
+		Device:      []string{s.device},
+		Format:      "UUID",
+		NoHeadings:  true,
+		Success:     &success,
+		Out:         &s.uuid,
+		ExecOptions: curveadm.ExecOptions(),
+	})
+	steps = append(steps, &step.Lambda{
+		Lambda: checkDeviceUUID(s.host, s.device, &success, &s.uuid),
+	})
+	steps = append(steps, &step.Lambda{ // generate record
+		Lambda: s.expression(&express2del, &express2add),
+	})
+
+	if len(*s.oldUUID) > 0 {
+		steps = append(steps, &step.Sed{ // remove old record
+			Files:       []string{os.GetFSTabPath()},
+			Expression:  &express2del,
+			InPlace:     true,
+			ExecOptions: curveadm.ExecOptions(),
+		})
+	}
+
+	for _, s := range steps {
+		err := s.Execute(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Execute ...
+func (s *deleteFSTab) Execute(ctx *context.Context) error {
+	// lock by memstorage
+	return s.curveadm.MemStorage().TX(func(m *utils.SafeMap) error {
+		return s.execute(ctx)
+	})
+}
+
+// stopContainer 停止container
+type stopContainer struct {
+	containerId *string
+	curveadm    *cli.CurveAdm
+}
+
+// Execute ...
+func (s *stopContainer) Execute(ctx *context.Context) error {
+	if len(*s.containerId) == 0 {
+		return nil
+	}
+
+	steps := []task.Step{}
+	steps = append(steps, &step.StopContainer{
+		ContainerId: *s.containerId,
+		Time:        1,
+		ExecOptions: s.curveadm.ExecOptions(),
+	})
+	for _, s := range steps {
+		err := s.Execute(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// NewStopFormatTask 新建停止stop format任务
+func NewStopFormatTask(curveadm *cli.CurveAdm, fc *configure.FormatConfig) (*task.Task, error) {
+	host := fc.GetHost()
+	hc, err := curveadm.GetHost(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// new task
+	device := fc.GetDevice()
+	mountPoint := fc.GetMountPoint()
+	usagePercent := fc.GetFormatPercent()
+	subname := fmt.Sprintf("host=%s device=%s mountPoint=%s usage=%d%%",
```

Comment:
Maybe the container id (or name) is more useful than the usage percent in stop task.

Reply:
> but we can't get containerid ,if task step not run?

There is a fixed method to generate container name for format task:

```
containerName := device2ContainerName(device)
```

Reply:
modify it ,replace the usagePercent with containerName